app/controllers/groupings_controller.rb

```diff
@@ -8,6 +12,19 @@ def show
     @background = @grouping.background.url
   end
 
+  def new
+    @grouping = Grouping.new
+  end
+
+  def create
+    @grouping = Grouping.new(new_grouping_params)
+    if @grouping.save
+      redirect_to @grouping
+    else
+      render 'new'
```

Comment:
Should we redirect instead of just rendering the `new` template?

Reply:
I don't think so, because then the user would lose the progress they made. If they have a mistake in one of the fields, we want the form to still be populated for them.